*** configure.ac	2017-01-03 10:53:52.436371005 +0000
--- ../../komodo-jl777/configure.ac	2017-01-03 09:49:08.848505929 +0000
***************
*** 124,130 ****
    [AS_HELP_STRING([--enable-hardening],
    [attempt to harden the resulting executables (default is yes)])],
    [use_hardening=$enableval],
!   [use_hardening=no])
  
  AC_ARG_ENABLE([reduce-exports],
    [AS_HELP_STRING([--enable-reduce-exports],
--- 124,130 ----
    [AS_HELP_STRING([--enable-hardening],
    [attempt to harden the resulting executables (default is yes)])],
    [use_hardening=$enableval],
!   [use_hardening=yes])
  
  AC_ARG_ENABLE([reduce-exports],
    [AS_HELP_STRING([--enable-reduce-exports],
***************
*** 276,305 ****
           fi
         fi
  
! dnl       AC_CHECK_PROG([BREW],brew, brew)
! dnl       if test x$BREW = xbrew; then
           dnl These Homebrew packages may be keg-only, meaning that they won't be found
           dnl in expected paths because they may conflict with system files. Ask
           dnl Homebrew where each one is located, then adjust paths accordingly.
           dnl It's safe to add these paths even if the functionality is disabled by
           dnl the user (--without-wallet or --without-gui for example).
  
! dnl         openssl_prefix=`$BREW --prefix openssl 2>/dev/null`
! dnl         bdb_prefix=`$BREW --prefix berkeley-db4 2>/dev/null`
! dnl         qt5_prefix=`$BREW --prefix qt5 2>/dev/null`
! dnl         if test x$openssl_prefix != x; then
! dnl           PKG_CONFIG_PATH="$PKG_CONFIG_PATH"
! dnl           export PKG_CONFIG_PATH
! dnl         fi
! dnl         if test x$bdb_prefix != x; then
! dnl           CPPFLAGS="$CPPFLAGS -I$bdb_prefix/include"
! dnl           LIBS="$LIBS -L$bdb_prefix/lib"
! dnl         fi
! dnl         if test x$qt5_prefix != x; then
! dnl           PKG_CONFIG_PATH="$qt5_prefix/lib/pkgconfig:$PKG_CONFIG_PATH"
! dnl           export PKG_CONFIG_PATH
! dnl         fi
! dnl       fi
       else
         case $build_os in
           *darwin*)
--- 276,305 ----
           fi
         fi
  
!        AC_CHECK_PROG([BREW],brew, brew)
!        if test x$BREW = xbrew; then
           dnl These Homebrew packages may be keg-only, meaning that they won't be found
           dnl in expected paths because they may conflict with system files. Ask
           dnl Homebrew where each one is located, then adjust paths accordingly.
           dnl It's safe to add these paths even if the functionality is disabled by
           dnl the user (--without-wallet or --without-gui for example).
  
!          openssl_prefix=`$BREW --prefix openssl 2>/dev/null`
!          bdb_prefix=`$BREW --prefix berkeley-db4 2>/dev/null`
!          qt5_prefix=`$BREW --prefix qt5 2>/dev/null`
!          if test x$openssl_prefix != x; then
!            PKG_CONFIG_PATH="$openssl_prefix/lib/pkgconfig:$PKG_CONFIG_PATH"
!            export PKG_CONFIG_PATH
!          fi
!          if test x$bdb_prefix != x; then
!            CPPFLAGS="$CPPFLAGS -I$bdb_prefix/include"
!            LIBS="$LIBS -L$bdb_prefix/lib"
!          fi
!          if test x$qt5_prefix != x; then
!            PKG_CONFIG_PATH="$qt5_prefix/lib/pkgconfig:$PKG_CONFIG_PATH"
!            export PKG_CONFIG_PATH
!          fi
!        fi
       else
         case $build_os in
           *darwin*)
***************
*** 730,736 ****
  AC_CHECK_HEADER([libsnark/gadgetlib1/gadget.hpp],,AC_MSG_ERROR(libsnark headers missing))
  AC_CHECK_LIB([snark],[main],LIBSNARK_LIBS=-lsnark, [AC_MSG_ERROR(libsnark missing)], [-lgmpxx])
  
! LIBZCASH_LIBS="-lsnark -lgmp -lgmpxx -lboost_system-mt -lcrypto -lsodium"
  
  CXXFLAGS_TEMP="$CXXFLAGS"
  LIBS_TEMP="$LIBS"
--- 730,736 ----
  AC_CHECK_HEADER([libsnark/gadgetlib1/gadget.hpp],,AC_MSG_ERROR(libsnark headers missing))
  AC_CHECK_LIB([snark],[main],LIBSNARK_LIBS=-lsnark, [AC_MSG_ERROR(libsnark missing)], [-lgmpxx])
  
! LIBZCASH_LIBS="-lsnark -lgmp -lgmpxx -lboost_system-mt -lcrypto -lsodium -fopenmp"
  
  CXXFLAGS_TEMP="$CXXFLAGS"
  LIBS_TEMP="$LIBS"
