*** depends/builders/darwin.mk	2017-01-03 10:53:52.440371182 +0000
--- ../../komodo-jl777/depends/builders/darwin.mk	2017-01-03 09:48:21.646034937 +0000
***************
*** 1,5 ****
! build_darwin_CC = gcc-6
! build_darwin_CXX = g++-6
  build_darwin_AR: = $(shell xcrun -f ar)
  build_darwin_RANLIB: = $(shell xcrun -f ranlib)
  build_darwin_STRIP: = $(shell xcrun -f strip)
--- 1,5 ----
! build_darwin_CC: = $(shell xcrun -f clang)
! build_darwin_CXX: = $(shell xcrun -f clang++)
  build_darwin_AR: = $(shell xcrun -f ar)
  build_darwin_RANLIB: = $(shell xcrun -f ranlib)
  build_darwin_STRIP: = $(shell xcrun -f strip)
***************
*** 7,17 ****
  build_darwin_NM: = $(shell xcrun -f nm)
  build_darwin_INSTALL_NAME_TOOL:=$(shell xcrun -f install_name_tool)
  build_darwin_SHA256SUM = shasum -a 256
! build_darwin_DOWNLOAD = wget --timeout=$(DOWNLOAD_CONNECT_TIMEOUT) --tries=$(DOWNLOAD_RETRIES) -nv -O
  
  #darwin host on darwin builder. overrides darwin host preferences.
! darwin_CC= gcc-6
! darwin_CXX= g++-6
  darwin_AR:=$(shell xcrun -f ar)
  darwin_RANLIB:=$(shell xcrun -f ranlib)
  darwin_STRIP:=$(shell xcrun -f strip)
--- 7,17 ----
  build_darwin_NM: = $(shell xcrun -f nm)
  build_darwin_INSTALL_NAME_TOOL:=$(shell xcrun -f install_name_tool)
  build_darwin_SHA256SUM = shasum -a 256
! build_darwin_DOWNLOAD = curl --connect-timeout $(DOWNLOAD_CONNECT_TIMEOUT) --retry $(DOWNLOAD_RETRIES) -L -o
  
  #darwin host on darwin builder. overrides darwin host preferences.
! darwin_CC=$(shell xcrun -f clang) -mmacosx-version-min=$(OSX_MIN_VERSION)
! darwin_CXX:=$(shell xcrun -f clang++) -mmacosx-version-min=$(OSX_MIN_VERSION)
  darwin_AR:=$(shell xcrun -f ar)
  darwin_RANLIB:=$(shell xcrun -f ranlib)
  darwin_STRIP:=$(shell xcrun -f strip)
