*** depends/hosts/darwin.mk	2017-01-03 10:53:52.440371182 +0000
--- ../../komodo-jl777/depends/hosts/darwin.mk	2017-01-03 09:48:21.650035146 +0000
***************
*** 2,9 ****
  OSX_SDK_VERSION=10.9
  OSX_SDK=$(SDK_PATH)/MacOSX$(OSX_SDK_VERSION).sdk
  LD64_VERSION=241.9
! darwin_CC=gcc-6 -target $(host) -mmacosx-version-min=$(OSX_MIN_VERSION) --sysroot $(OSX_SDK) -mlinker-version=$(LD64_VERSION)
! darwin_CXX=g++-6 -target $(host) -mmacosx-version-min=$(OSX_MIN_VERSION) --sysroot $(OSX_SDK) -mlinker-version=$(LD64_VERSION)
  
  darwin_CFLAGS=-pipe
  darwin_CXXFLAGS=$(darwin_CFLAGS)
--- 2,9 ----
  OSX_SDK_VERSION=10.9
  OSX_SDK=$(SDK_PATH)/MacOSX$(OSX_SDK_VERSION).sdk
  LD64_VERSION=241.9
! darwin_CC=clang -target $(host) -mmacosx-version-min=$(OSX_MIN_VERSION) --sysroot $(OSX_SDK) -mlinker-version=$(LD64_VERSION)
! darwin_CXX=clang++ -target $(host) -mmacosx-version-min=$(OSX_MIN_VERSION) --sysroot $(OSX_SDK) -mlinker-version=$(LD64_VERSION)
  
  darwin_CFLAGS=-pipe
  darwin_CXXFLAGS=$(darwin_CFLAGS)
