*** src/crypto/equihash.cpp	2017-01-03 10:53:52.444371361 +0000
--- ../../komodo-jl777/src/crypto/equihash.cpp	2017-01-03 09:48:21.690037241 +0000
***************
*** 21,50 ****
  
  #include <boost/optional.hpp>
  
- #ifdef __APPLE__
- #include <machine/endian.h>
- #include <libkern/OSByteOrder.h>
- 
- #define htobe16(x) OSSwapHostToBigInt16(x)
- #define htole16(x) OSSwapHostToLittleInt16(x)
- #define be16toh(x) OSSwapBigToHostInt16(x)
- #define le16toh(x) OSSwapLittleToHostInt16(x)
- 
- #define htobe32(x) OSSwapHostToBigInt32(x)
- #define htole32(x) OSSwapHostToLittleInt32(x)
- #define be32toh(x) OSSwapBigToHostInt32(x)
- #define le32toh(x) OSSwapLittleToHostInt32(x)
- 
- #define htobe64(x) OSSwapHostToBigInt64(x)
- #define htole64(x) OSSwapHostToLittleInt64(x)
- #define be64toh(x) OSSwapBigToHostInt64(x)
- #define le64toh(x) OSSwapLittleToHostInt64(x)
- 
- #define __BIG_ENDIAN    BIG_ENDIAN
- #define __LITTLE_ENDIAN LITTLE_ENDIAN
- #define __BYTE_ORDER    BYTE_ORDER
- #endif
- 
  EhSolverCancelledException solver_cancelled;
  
  template<unsigned int N, unsigned int K>
--- 21,26 ----
