*** depends/packages/libsnark.mk	2017-01-03 10:53:52.440371182 +0000
--- ../../komodo-jl777/depends/packages/libsnark.mk	2017-01-03 09:48:21.650035146 +0000
***************
*** 1,30 ****
  package=libsnark
  $(package)_version=0.1
! $(package)_download_path=https://github.com/radix42/$(package)/archive
  $(package)_file_name=$(package)-$($(package)_git_commit).tar.gz
  $(package)_download_file=$($(package)_git_commit).tar.gz
! $(package)_sha256_hash=9dbd5b44d3443e86463e934bfe1023cab4ca5948f8d74c23a67d9535c28d2584
! $(package)_git_commit=9be18569b8abcda1245c3912877075259599c0f1
  
  $(package)_dependencies=libgmp libsodium
  
- ifeq ($(build_os),darwin)
- define $(package)_build_cmds
- CC=gcc-6 CXX=g++-6 CXXFLAGS="-arch x86_64 -DBINARY_OUTPUT -DNO_PT_COMPRESSION=1" $(MAKE) lib DEPINST=$(host_prefix) CURVE=ALT_BN128 MULTICORE=0 NO_PROCPS=1 NO_GTEST=1 NO_DOCS=1 STATIC=1 NO_SUPERCOP=1 FEATUREFLAGS=-DMONTGOMERY_OUTPUT
- endef
- else
  define $(package)_build_cmds
    CXXFLAGS="-fPIC -DBINARY_OUTPUT -DNO_PT_COMPRESSION=1" $(MAKE) lib DEPINST=$(host_prefix) CURVE=ALT_BN128 MULTICORE=1 NO_PROCPS=1 NO_GTEST=1 NO_DOCS=1 STATIC=1 NO_SUPERCOP=1 FEATUREFLAGS=-DMONTGOMERY_OUTPUT OPTFLAGS="-O2 -march=x86-64 -g "
  endef
  
- 
- 
- 
- endif
- 
- 
- 
- 
  define $(package)_stage_cmds
    $(MAKE) install STATIC=1 DEPINST=$(host_prefix) PREFIX=$($(package)_staging_dir)$(host_prefix) CURVE=ALT_BN128 NO_SUPERCOP=1
  endef
--- 1,17 ----
  package=libsnark
  $(package)_version=0.1
! $(package)_download_path=https://github.com/zcash/$(package)/archive
  $(package)_file_name=$(package)-$($(package)_git_commit).tar.gz
  $(package)_download_file=$($(package)_git_commit).tar.gz
! $(package)_sha256_hash=9422b1a2a94e6b8be61f07af7f146087c2a7d70b208d07ad076622225aa7f0e4
! $(package)_git_commit=2e6314a9f7efcd9af1c77669d7d9a229df86a777
  
  $(package)_dependencies=libgmp libsodium
  
  define $(package)_build_cmds
    CXXFLAGS="-fPIC -DBINARY_OUTPUT -DNO_PT_COMPRESSION=1" $(MAKE) lib DEPINST=$(host_prefix) CURVE=ALT_BN128 MULTICORE=1 NO_PROCPS=1 NO_GTEST=1 NO_DOCS=1 STATIC=1 NO_SUPERCOP=1 FEATUREFLAGS=-DMONTGOMERY_OUTPUT OPTFLAGS="-O2 -march=x86-64 -g "
  endef
  
  define $(package)_stage_cmds
    $(MAKE) install STATIC=1 DEPINST=$(host_prefix) PREFIX=$($(package)_staging_dir)$(host_prefix) CURVE=ALT_BN128 NO_SUPERCOP=1
  endef
