*** Makefile.am	2017-01-03 10:53:52.436371005 +0000
--- ../../komodo-jl777/Makefile.am	2017-01-03 09:49:08.848505929 +0000
***************
*** 39,58 ****
    $(top_srcdir)/contrib/macdeploy/detached-sig-apply.sh \
    $(top_srcdir)/contrib/macdeploy/detached-sig-create.sh
  
- 
- if BUILD_DARWIN
- COVERAGE_INFO = baseline_filtered_combined.info baseline.info block_test.info \
-   leveldb_baseline.info test_bitcoin_filtered.info total_coverage.info \
-   baseline_filtered.info block_test_filtered.info \
-   leveldb_baseline_filtered.info test_bitcoin_coverage.info test_bitcoin.info
- #  zcash-gtest.info zcash-gtest_filtered.info zcash-gtest_coverage.info
- else
  COVERAGE_INFO = baseline_filtered_combined.info baseline.info block_test.info \
    leveldb_baseline.info test_bitcoin_filtered.info total_coverage.info \
    baseline_filtered.info block_test_filtered.info \
    leveldb_baseline_filtered.info test_bitcoin_coverage.info test_bitcoin.info \
    zcash-gtest.info zcash-gtest_filtered.info zcash-gtest_coverage.info
- endif
  
  dist-hook:
  	-$(MAKE) -C $(top_distdir)/src/leveldb clean
--- 39,49 ----
***************
*** 159,173 ****
  baseline.info:
  	$(LCOV) -c -i -d $(abs_builddir)/src -o $@
  
- if BUILD_DARWIN
- baseline_filtered.info: baseline.info
- 	$(LCOV) -r $< "/usr/include/*" \
-                       "$(abs_builddir)/depends/x86_64-unknown-linux-gnu/include/*.h" \
-                       "$(abs_builddir)/depends/x86_64-unknown-linux-gnu/include/boost/*" \
-                       "$(abs_builddir)/depends/x86_64-unknown-linux-gnu/include/gmock/*" \
-                       "$(abs_builddir)/src/test/*" \
-                       -o $@
- else
  baseline_filtered.info: baseline.info
  	$(LCOV) -r $< "/usr/include/*" \
                        "$(abs_builddir)/depends/x86_64-unknown-linux-gnu/include/*.h" \
--- 150,155 ----
***************
*** 177,196 ****
                        "$(abs_builddir)/src/gtest/*" \
                        "$(abs_builddir)/src/test/*" \
                        -o $@
- endif
  
  leveldb_baseline.info: baseline_filtered.info
  	$(LCOV) -c -i -d $(abs_builddir)/src/leveldb -b $(abs_builddir)/src/leveldb -o $@
  
- if BUILD_DARWIN
- leveldb_baseline_filtered.info: leveldb_baseline.info
- 	$(LCOV) -r $< "/usr/include/*" \
-                       "$(abs_builddir)/depends/x86_64-unknown-linux-gnu/include/*.h" \
-                       "$(abs_builddir)/depends/x86_64-unknown-linux-gnu/include/boost/*" \
-                       "$(abs_builddir)/depends/x86_64-unknown-linux-gnu/include/gmock/*" \
-                       "$(abs_builddir)/src/test/*" \
-                       -o $@
- else
  leveldb_baseline_filtered.info: leveldb_baseline.info
  	$(LCOV) -r $< "/usr/include/*" \
                        "$(abs_builddir)/depends/x86_64-unknown-linux-gnu/include/*.h" \
--- 159,168 ----
***************
*** 200,206 ****
                        "$(abs_builddir)/src/gtest/*" \
                        "$(abs_builddir)/src/test/*" \
                        -o $@
- endif
  
  baseline_filtered_combined.info: leveldb_baseline_filtered.info baseline_filtered.info
  	$(LCOV) -a leveldb_baseline_filtered.info -a baseline_filtered.info -o $@
--- 172,177 ----
***************
*** 211,226 ****
  	$(LCOV) -z -d $(abs_builddir)/src
  	$(LCOV) -z -d $(abs_builddir)/src/leveldb
  
- if BUILD_DARWIN
  test_bitcoin_filtered.info: test_bitcoin.info
  	$(LCOV) -r $< "/usr/include/*" \
                        "$(abs_builddir)/depends/x86_64-unknown-linux-gnu/include/*.h" \
                        "$(abs_builddir)/depends/x86_64-unknown-linux-gnu/include/boost/*" \
                        "$(abs_builddir)/depends/x86_64-unknown-linux-gnu/include/gmock/*" \
                        "$(abs_builddir)/src/test/*" \
                        -o $@
! else
! test_bitcoin_filtered.info: test_bitcoin.info
  	$(LCOV) -r $< "/usr/include/*" \
                        "$(abs_builddir)/depends/x86_64-unknown-linux-gnu/include/*.h" \
                        "$(abs_builddir)/depends/x86_64-unknown-linux-gnu/include/boost/*" \
--- 182,204 ----
  	$(LCOV) -z -d $(abs_builddir)/src
  	$(LCOV) -z -d $(abs_builddir)/src/leveldb
  
  test_bitcoin_filtered.info: test_bitcoin.info
  	$(LCOV) -r $< "/usr/include/*" \
                        "$(abs_builddir)/depends/x86_64-unknown-linux-gnu/include/*.h" \
                        "$(abs_builddir)/depends/x86_64-unknown-linux-gnu/include/boost/*" \
                        "$(abs_builddir)/depends/x86_64-unknown-linux-gnu/include/gmock/*" \
+                       "$(abs_builddir)/depends/x86_64-unknown-linux-gnu/include/gtest/*" \
+                       "$(abs_builddir)/src/gtest/*" \
                        "$(abs_builddir)/src/test/*" \
                        -o $@
! 
! zcash-gtest.info: baseline_filtered_combined.info
! 	$(MAKE) -C src/ zcash-gtest_check
! 	$(LCOV) -c -d $(abs_builddir)/src -t zcash-gtest -o $@
! 	$(LCOV) -z -d $(abs_builddir)/src
! 	$(LCOV) -z -d $(abs_builddir)/src/leveldb
! 
! zcash-gtest_filtered.info: zcash-gtest.info
  	$(LCOV) -r $< "/usr/include/*" \
                        "$(abs_builddir)/depends/x86_64-unknown-linux-gnu/include/*.h" \
                        "$(abs_builddir)/depends/x86_64-unknown-linux-gnu/include/boost/*" \
***************
*** 229,235 ****
                        "$(abs_builddir)/src/gtest/*" \
                        "$(abs_builddir)/src/test/*" \
                        -o $@
- endif
  
  block_test.info: test_bitcoin_filtered.info
  	$(MKDIR_P) qa/tmp
--- 207,212 ----
***************
*** 238,252 ****
  	$(LCOV) -z -d $(abs_builddir)/src
  	$(LCOV) -z -d $(abs_builddir)/src/leveldb
  
- if BUILD_DARWIN
- block_test_filtered.info: block_test.info
- 	$(LCOV) -r $< "/usr/include/*" \
-                       "$(abs_builddir)/depends/x86_64-unknown-linux-gnu/include/*.h" \
-                       "$(abs_builddir)/depends/x86_64-unknown-linux-gnu/include/boost/*" \
-                       "$(abs_builddir)/depends/x86_64-unknown-linux-gnu/include/gmock/*" \
-                       "$(abs_builddir)/src/test/*" \
-                       -o $@
- else
  block_test_filtered.info: block_test.info
  	$(LCOV) -r $< "/usr/include/*" \
                        "$(abs_builddir)/depends/x86_64-unknown-linux-gnu/include/*.h" \
--- 215,220 ----
***************
*** 256,293 ****
                        "$(abs_builddir)/src/gtest/*" \
                        "$(abs_builddir)/src/test/*" \
                        -o $@
- endif
  
  test_bitcoin_coverage.info: baseline_filtered_combined.info test_bitcoin_filtered.info
!        $(LCOV) -a baseline_filtered.info -a leveldb_baseline_filtered.info -a test_bitcoin_filtered.info -o $@
  
- if ! BUILD_DARWIN
  zcash-gtest_coverage.info: baseline_filtered_combined.info zcash-gtest_filtered.info
  	$(LCOV) -a baseline_filtered.info -a leveldb_baseline_filtered.info -a zcash-gtest_filtered.info -o $@
- endif
  
- if BUILD_DARWIN
- total_coverage.info:  baseline_filtered_combined.info test_bitcoin_filtered.info block_test_filtered.info
-        $(LCOV) -a baseline_filtered.info -a leveldb_baseline_filtered.info -a test_bitcoin_filtered.info -a block_test_filtered.info -o $@ | $(GREP) "\%" | $(AWK) '{ print substr($$3,2,50) "/" $$5 }' > coverage_percent.txt
- else
  total_coverage.info:  baseline_filtered_combined.info test_bitcoin_filtered.info zcash-gtest_filtered.info block_test_filtered.info
  	$(LCOV) -a baseline_filtered.info -a leveldb_baseline_filtered.info -a test_bitcoin_filtered.info -a zcash-gtest_filtered.info -a block_test_filtered.info -o $@ | $(GREP) "\%" | $(AWK) '{ print substr($$3,2,50) "/" $$5 }' > coverage_percent.txt
- endif
- 
  
  test_bitcoin.coverage/.dirstamp:  test_bitcoin_coverage.info
!        $(GENHTML) -s $< -o $(@D)
!        @touch $@
  
  total.coverage/.dirstamp: total_coverage.info
!        $(GENHTML) -s $< -o $(@D)
!        @touch $@
  
- if BUILD_DARWIN
- cov: test_bitcoin.coverage/.dirstamp total.coverage/.dirstamp
- else
  cov: test_bitcoin.coverage/.dirstamp cov-zcash total.coverage/.dirstamp
- endif
  
  endif
  
--- 224,254 ----
                        "$(abs_builddir)/src/gtest/*" \
                        "$(abs_builddir)/src/test/*" \
                        -o $@
  
  test_bitcoin_coverage.info: baseline_filtered_combined.info test_bitcoin_filtered.info
! 	$(LCOV) -a baseline_filtered.info -a leveldb_baseline_filtered.info -a test_bitcoin_filtered.info -o $@
  
  zcash-gtest_coverage.info: baseline_filtered_combined.info zcash-gtest_filtered.info
  	$(LCOV) -a baseline_filtered.info -a leveldb_baseline_filtered.info -a zcash-gtest_filtered.info -o $@
  
  total_coverage.info:  baseline_filtered_combined.info test_bitcoin_filtered.info zcash-gtest_filtered.info block_test_filtered.info
  	$(LCOV) -a baseline_filtered.info -a leveldb_baseline_filtered.info -a test_bitcoin_filtered.info -a zcash-gtest_filtered.info -a block_test_filtered.info -o $@ | $(GREP) "\%" | $(AWK) '{ print substr($$3,2,50) "/" $$5 }' > coverage_percent.txt
  
  test_bitcoin.coverage/.dirstamp:  test_bitcoin_coverage.info
! 	$(GENHTML) -s $< -o $(@D)
! 	@touch $@
! 
! zcash-gtest.coverage/.dirstamp:  zcash-gtest_coverage.info
! 	$(GENHTML) -s $< -o $(@D)
! 	@touch $@
! 
! cov-zcash: zcash-gtest.coverage/.dirstamp
  
  total.coverage/.dirstamp: total_coverage.info
! 	$(GENHTML) -s $< -o $(@D)
! 	@touch $@
  
  cov: test_bitcoin.coverage/.dirstamp cov-zcash total.coverage/.dirstamp
  
  endif
  
