diff -ur zeromq-4.3.3-orig/src/clock.cpp zeromq-4.3.3/src/clock.cpp
--- zeromq-4.3.3-orig/src/clock.cpp	2020-10-13 13:44:04.190000000 +0100
+++ zeromq-4.3.3/src/clock.cpp	2020-10-13 13:47:04.170000000 +0100
@@ -126,9 +126,13 @@
   init_compatible_get_tick_count64 ();
 #endif
 
+#ifndef ZMQ_HAVE_WINDOWS
 const uint64_t usecs_per_msec = 1000;
+#endif
 const uint64_t usecs_per_sec = 1000000;
+#ifndef ZMQ_HAVE_WINDOWS
 const uint64_t nsecs_per_usec = 1000;
+#endif
 
 zmq::clock_t::clock_t () :
     _last_tsc (rdtsc ()),
