diff --git a/zcutil/fetch-params.sh b/zcutil/fetch-params.sh
index b60fecb63..1d3d4f834 100755
--- a/zcutil/fetch-params.sh
+++ b/zcutil/fetch-params.sh
@@ -32,7 +32,7 @@ ZC_DISABLE_WGET="${ZC_DISABLE_WGET:-}"
 ZC_DISABLE_IPFS="${ZC_DISABLE_IPFS:-}"
 ZC_DISABLE_CURL="${ZC_DISABLE_CURL:-}"
 
-LOCKFILE=/tmp/fetch_params.lock
+LOCKFILE=$HOME/fetch_params.lock
 
 fetch_wget() {
     if [ -z "$WGETCMD" ] || [ -n "$ZC_DISABLE_WGET" ]; then
@@ -171,6 +171,10 @@ exit_locked_error() {
 }
 
 main() {
+    if ! [ -d "$PARAMS_DIR" ]
+    then
+        mkdir -p "$PARAMS_DIR"
+    fi
 
     lock fetch-params.sh \
     || exit_locked_error
@@ -185,20 +189,17 @@ If they already exist locally, it will exit now and do nothing else.
 EOF
 
     # Now create PARAMS_DIR and insert a README if necessary:
-    if ! [ -d "$PARAMS_DIR" ]
-    then
-        mkdir -p "$PARAMS_DIR"
-        README_PATH="$PARAMS_DIR/README"
-        cat >> "$README_PATH" <<EOF
+    README_PATH="$PARAMS_DIR/README"
+    cat >> "$README_PATH" <<EOF
 This directory stores common Zcash zkSNARK parameters. Note that it is
 distinct from the daemon's -datadir argument because the parameters are
 large and may be shared across multiple distinct -datadir's such as when
 setting up test networks.
 EOF
 
-        # This may be the first time the user's run this script, so give
-        # them some info, especially about bandwidth usage:
-        cat <<EOF
+    # This may be the first time the user's run this script, so give
+    # them some info, especially about bandwidth usage:
+    cat <<EOF
 The complete parameters are currently just under 1.7GB in size, so plan 
 accordingly for your bandwidth constraints. If the Sprout parameters are
 already present the additional Sapling parameters required are just under 
@@ -209,7 +210,6 @@ Creating params directory. For details about this directory, see:
 $README_PATH
 
 EOF
-    fi
 
     cd "$PARAMS_DIR"
 
