diff --git a/tinyformat.h b/tinyformat.h
index 7c607d7f0..1347832b2 100644
--- a/tinyformat.h
+++ b/tinyformat.h
@@ -1033,6 +1033,17 @@ TINYFORMAT_FOREACH_ARGNUM(TINYFORMAT_MAKE_FORMAT_FUNCS)
 
 #endif
 
+// Added for Bitcoin Core
+template<typename... Args>
+std::string format(const std::string &fmt, const Args&... args)
+{
+    std::ostringstream oss;
+    format(oss, fmt.c_str(), args...);
+    return oss.str();
+}
+
 } // namespace tinyformat
 
+#define strprintf tfm::format
+
 #endif // TINYFORMAT_H_INCLUDED
