commit 871dac7fcea001b9c7f88a86fd6cce424ebeb068
Author: Nathan Wilcox <nejucomo@gmail.com>
Date:   Tue May 12 14:07:26 2015 -0700

    Do not assume DEPINST is in use during install.

diff --git a/Makefile b/Makefile
index 93ef3b2..c2226a0 100644
--- a/Makefile
+++ b/Makefile
@@ -193,8 +193,8 @@ $(HEADERS_DEST): $(PREFIX)/include/libsnark/%: src/%
 install: lib $(HEADERS_DEST)
 	mkdir -p $(PREFIX)/lib
 	cp $(LIBOBJ) $(PREFIX)/lib/$(LIBOBJ)
-	cp -rv $(DEPINST)/lib $(PREFIX)
-	cp -rv $(DEPINST)/include $(PREFIX)
+	[ \! -d $(DEPINST)/lib ] || cp -rv $(DEPINST)/lib $(PREFIX)
+	[ \! -d $(DEPINST)/include ] || cp -rv $(DEPINST)/include $(PREFIX)
 endif
 
 doxy:
