commit 85e660ec91d4fce84a6e1eb1b5ad757d6d645344
Author: Taylor Hornby <havoc@defuse.ca>
Date:   Wed Sep 9 16:48:36 2015 -0600

    Do not assume DEPINST is in use during install.

diff --git a/Makefile b/Makefile
index daf7d26..cb9b751 100644
--- a/Makefile
+++ b/Makefile
@@ -247,8 +247,8 @@ $(HEADERS_DEST): $(PREFIX)/include/libsnark/%: src/%
 install: lib $(HEADERS_DEST)
 	mkdir -p $(PREFIX)/lib
 	cp $(LIBOBJ) $(PREFIX)/lib/$(LIBOBJ)
-	cp -rv $(DEPINST)/lib $(PREFIX)
-	cp -rv $(DEPINST)/include $(PREFIX)
+	[ \! -d $(DEPINST)/lib ] || cp -rv $(DEPINST)/lib $(PREFIX)
+	[ \! -d $(DEPINST)/include ] || cp -rv $(DEPINST)/include $(PREFIX)
 endif
 
 doxy:
