diff --git a/libzerocash/Address.cpp b/libzerocash/Address.cpp
index f0b7f3c..a6b642d 100644
--- a/libzerocash/Address.cpp
+++ b/libzerocash/Address.cpp
@@ -106,6 +106,13 @@ bool PublicAddress::operator!=(const PublicAddress& rhs) const {
 	return !(*this == rhs);
 }
 
+Address::Address(PrivateAddress& priv) {
+    addr_sk = priv;
+
+    PublicAddress pubaddr(priv.getAddressSecret(), priv.getEncryptionSecretKey());
+
+    addr_pk = pubaddr;
+}
 
 Address::Address(): addr_pk(), addr_sk() {
     std::vector<unsigned char> a_sk(a_sk_size);
diff --git a/libzerocash/Address.h b/libzerocash/Address.h
index 2ad029c..93c3378 100644
--- a/libzerocash/Address.h
+++ b/libzerocash/Address.h
@@ -93,6 +93,7 @@ friend class PourTransaction;
 
 public:
 	Address();
+	Address(PrivateAddress&);
 
 	const PublicAddress& getPublicAddress() const;
 	const PrivateAddress& getPrivateAddress() const;
